### PR TITLE
[FIX] Improve parallel theme request handling

### PR DESCRIPTION
**Pull request by matz3**

### CODE REVIEW

File: lib/middleware/serveThemes.js [hunk 1]
```diff
@@ -45,6 +45,54 @@ function createMiddleware({resources, middlewareUtil}) {
...
+		if (!sourceLessResource) { // Not found
+			return;
+		}
+
+		const createdResources = await builder.build([sourceLessResource], buildOptions);
+
+		// Pick requested file resource
+		const resource = createdResources.find((res) => res.getPath().endsWith(filename));
+		if (!resource) {
+			throw new Error(`Theme Build did not return requested file "${pathname}"`);
```
Comment (codeworrior) [inline lib/middleware/serveThemes.js:71]: is `endsWith` appropriate? Shouldn't this be a call to `basename(res.getPath()) === filename`?

**Review by svbender — COMMENTED** [review]
Looks good. Maybe you want to add the improvement as codeworrior suggested?

File: lib/middleware/serveThemes.js (hunk 1 above)
Comment (matz3) [reply lib/middleware/serveThemes.js:71]: Done

**Review by svbender — APPROVED** [review]
Nice!

**Pull request merged**